lib/deref.js

```diff
@@ -383,5 +389,67 @@ module.exports = {
     }
 
     return schema;
+  },
+
+  /**
+   * Take a $ref and a spec and return the referenced value
+   * @param {object} spec the parsed spec object
+   * @param {string} reference the $ref value to dereference
+   * @returns {object} the dereferenced $ref value
+   */
+  dereferenceElement: function (spec, reference) {
```

Comment:
maybe reference could be renamed to referencePointer or jsonPointer, what do you think?